src/Sparrow.Server/Allocators.Arena.cs

```diff
@@ -0,0 +1,345 @@
+            {
+                // Override with the current setup
+                preferredSize = allocator._currentBuffer.Size;
+            }
+
+            if (allocator._currentBuffer.IsValid)
+            {
+                allocator.Allocated -= _currentBuffer.Size;
+                allocator._nativeAllocator.Release(ref allocator._nativeAllocator, ref _currentBuffer);                
+            }                
+
+            // We will always cap the size of the Arena toward the MaxArenaSize.
+            if (preferredSize > allocator._options.MaxArenaSize)
+                preferredSize = allocator._options.MaxArenaSize;
+
+            allocator._preferredSize = preferredSize;
+            allocator._currentBuffer = new Pointer();
+            allocator._ptrStart = allocator._ptrCurrent = null;
+            allocator._used = 0;
+
+            allocator.TotalAllocated = 0;
+            allocator.Allocated = 0;
+            allocator.InUse = 0;
+        }
+
+        public void Renew(ref ArenaAllocator<TOptions> allocator)
+        {
+            if (!allocator._currentBuffer.IsValid)
+            {
+                allocator._currentBuffer = allocator._nativeAllocator.Allocate(ref allocator._nativeAllocator, (int)allocator._preferredSize);
+                allocator.Allocated += allocator._currentBuffer.Size;
+            }
+
+            if (allocator._olderBuffers != null)
+            {
+                // Free old buffers not being used anymore. A single arena ought to be enough for anybody, you know ;) 
+                foreach (var unusedBuffer in allocator._olderBuffers)
+                {
+                    _used += unusedBuffer.Size;
+
+                    Pointer ptr = unusedBuffer;
+
+                    allocator.Allocated -= ptr.Size;
+                    allocator._nativeAllocator.Release(ref allocator._nativeAllocator, ref ptr);
+                }
+
+                allocator._olderBuffers.Clear();
+            }
+
+            allocator._ptrStart = allocator._ptrCurrent = (byte*)allocator._currentBuffer.Address;
+            allocator._used = 0;
+            allocator.InUse = 0;
+        }
+
+        public void OnAllocate(ref ArenaAllocator<TOptions> allocator, Pointer ptr)
+        {
+            // This allocator does not keep track of anything.
+        }
+
+        public void OnRelease(ref ArenaAllocator<TOptions> allocator, Pointer ptr)
+        {
+            // This allocator does not keep track of anything.
+        }
+
+        public void Dispose(ref ArenaAllocator<TOptions> allocator, bool disposing)
+        {            
+            if (allocator._olderBuffers != null)
+            {
+                // Free the remnant buffers on local storage from last cycle regrowth
+                foreach (var unusedBuffer in allocator._olderBuffers)
+                {
+                    _used += unusedBuffer.Size;
```

Comment:
Actually, that is a no-op bug translated because of copy-pasta from Reset and Renew where it is added in the loop and then setup to 0 at the end of the method. Good catch. Removing from the 3 sites.